src/client/mod.rs

```diff
@@ -290,464 +215,122 @@ impl Client {
         Ok(Client {
             flags: flags,
             cluster_id: cluster_id,
-            mmv_path: mmv_path,
-            wi: MMVWriterInfo::new()
+            mmv_path: mmv_path
         })
     }
+    
+    pub fn export(&self, metrics: &mut [&mut MMVWriter]) -> io::Result<()> {
```

Comment:
> The mutable references live only within the block of export, so metrics can be freely mutated after the call.

This works only if there is a 1:1 relationship between a metric and a client. There might be an application creating 2 clients that might want to share a metric. In that case, if both export concurrently, I don't think both will be able to get a mutable reference to the same metric (from my understanding)



> Also not sure why to limit the client to export only once? If the user wanted to add/remove metrics under the same name or cluster ID, it would be the only way to do it. Doesn't Speed also allow starting/stopping the mmap multiple times?

Yeah, sorry missed that. My bad. Please add this behavior in the documentation for `export`